anesthetic/plot.py

```diff
@@ -402,6 +392,79 @@ def contour_plot_2d(ax, data_x, data_y, *args, **kwargs):
     return cbar
 
 
+def hist_plot_2d(ax, data_x, data_y, *args, **kwargs):
+    """Plot a 2d marginalised distribution as a histogram.
+
+    This functions as a wrapper around matplotlib.axes.Axes.hist2d
+
+    Parameters
+    ----------
+    ax: matplotlib.axes.Axes
+        axis object to plot on
+
+    data_x, data_y: numpy.array
+        x and y coordinates of uniformly weighted samples to generate kernel
+        density estimator.
+
+    xmin, xmax, ymin, ymax: float
+        lower/upper prior bounds in x/y coordinates
+        optional, default None
+
+    contours: list
+        amount of mass within each iso-probability contour.
+        If None defaults to hist2d.
+        Default [0.68, 0.95]
+
+    Returns
+    -------
+    c: matplotlib.collections.QuadMesh
+        A set of colors
+
+    """
+    xmin = kwargs.pop('xmin', None)
+    xmax = kwargs.pop('xmax', None)
+    ymin = kwargs.pop('ymin', None)
+    ymax = kwargs.pop('ymax', None)
+    label = kwargs.pop('label', None)
+    contours = kwargs.pop('contours', [0.68, 0.95])
+    color = kwargs.pop('color', next(ax._get_lines.prop_cycler)['color'])
+
+    if len(data_x) == 0 or len(data_y) == 0:
+        return numpy.zeros(0), numpy.zeros(0), numpy.zeros((0, 0))
+
+    cmap = basic_cmap(color)
+
+    if contours is None:
+        pdf, x, y, image = ax.hist2d(data_x, data_y, cmap=cmap,
```

Comment:
Since `contours` is kwarg popped earlier, this option only gets triggered when one expressedly sets `contours` to `None`. From the docs I take it that that is the intended behaviour?

For me the contour-hist would not be what I would go for. I typically go for a hist to play around with the number of bins and get a feel for the data. But the contour-hist seems to me analogous to in the 1D case switching from a KDE to a hist with just 3 bins. But since it's in the docs, I guess it doesn't matter terribly much.

Reply:
￼8fd14ca adds the contours kwarg to ``contour_plot_2d``, so that it's consistent with ``hist_2d``. I have two questions
1. Should the default be `[0.68, 0.95]`, or `None`, for either `hist_2d` or `contour_plot_2d`
2. Should we use the keyword `contours`, or `levels`. The latter is the argument that is passed into `contour` and `contourf`, so would avoid conflicts.

Reply:
These questions address two objectives of this package that sometimes point in different directions: 
On the one hand we want to provide a nice tool for plotting MCMC and Neste Sampling output. On the other hand we want a package that nicely integrates with matplotlib and thus making it intuitively usable for people familiar with matplotlib.

1. Here, I feel like the defaults would indeed be different for `hist_2d` and `contour_plot_2d`.
    - `contour_plot_2d`: For contour plots we definitely want the 68 and 95% contours as default. This is the standard which people will want to plot. Hence, I'd go for either `[0.68, 0.95]` as default for `contours` or `[0.32, 0.05]` as default for `levels`.
    - `hist_2d`: I would swap to 2D histograms when I want a more fine grained view of the data and play around with how different bins might affect the picture. For that I wouldn't want to see only 2 quantiles, I would want a proper histogram. Also for flat priors 2D histograms might be useful, but also in this case I wouldn't want quantiles but a proper histogram that shows that the data is evenly spread across all bins. Hence, for `hist_2d` I would prefer the default set to `None`.

2. A case can be made for either one here. Do we want to stick close to `matplotlib` or change things to be closer to the usage.
    - `contours` is the term typically used in the context of these plots ("68 and 95% contours"). It would also indicate that these are not the `matplotlib` typical contour plots, but are designed to be blobs/bananas/... with about 2 levels.
    - `levels` is the term that's more conform with `matplotlib`. This would prevent having two terms for essentially the same thing which would simplify the kwarg handling. I think I personally am leaning to this option, but if we use this we probably should document that levels `[0.32, 0.05]` correspond to 68 and 95% contours. Or should we invert the levels internally, i.e. `levels = 1. - levels`?

Reply:
I agree with point 1. Now that `contour_plot_2d` has a `contours`/`levels` kwarg, it's more obvious that you can supply these to hist_2d if desired. Changes pushed

I've opted for levels, with the defaults at the sensible values [0.68, 0.95].